src/coreclr/jit/importercalls.cpp

```diff
@@ -191,10 +191,15 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
             }
         }
 
-        clsHnd = pResolvedToken->hClass;
-
+        clsHnd   = pResolvedToken->hClass;
         clsFlags = callInfo->classFlags;
 
+        // if clsHnd is an interface and method is static, try to get the actual implementation class
+        if ((clsFlags & CORINFO_FLG_INTERFACE) != 0 && (clsFlags & CORINFO_FLG_INTERFACE) != 0)
```

Comment:
There are other situations where `getCallInfo` can resolve the method on a different class. Would it be better to return the class handle of the resolved method in `CORINFO_CALL_INFO`?

Reply:
(Also, the condition has typo - it checks `CORINFO_FLG_INTERFACE` twice.)

Reply:
@jkotas do you mean getCallInfo should patch input resolveToken->hClass? Because it seems that JIT mostly access hClass directly from resolveToken for methods in importer

Reply:
I meant that there should be a new `CORINFO_CLASS_HANDLE hClass; //target class handle` field in `CORINFO_CALL_INFO`.

Patching `CORINFO_RESOLVED_TOKEN` would be a hack. `CORINFO_RESOLVED_TOKEN` is meant to be "in" argument for `getCallInfo`.

> it seems that JIT mostly access hClass directly from resolveToken for methods in importer

I would expect that some places want the `hClass` of the method referenced by the IL and some places want the `hClass` of the method that the VM resolved the method to. I would not be surprised if there are subtle bugs today like the one you are trying to fix.